include/eve/detail/function/gather_high.hpp

```diff
@@ -0,0 +1,32 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2020 Joel FALCOU
+  Copyright 2020 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/overload.hpp>
+#include <eve/concept/vectorized.hpp>
+
+namespace eve
+{
+  EVE_MAKE_CALLABLE(gather_high_, gather_high);
+
+  // Acts as a pattern itself
+  template<> struct is_pattern<tag::gather_high_> : std::true_type {};
+
+  // Premade pattern generator
+  template<std::ptrdiff_t N>
+  inline constexpr auto gather_high_n = fix_pattern<N>([](auto i, auto c) { return c/2 + i%(c/2); });
+}
```

Comment:
Why gather high?

Hmm.

No, I don't understand.

For 8 elements:

```
4, 5, 6, 7, 4, 5, 6, 7
```
?

As we discussed I really think this should just bulk up to half the register size.
But ok.